packages/darklang/languageTools/parser.dark

```diff
@@ -554,6 +554,73 @@ module Darklang =
                 (WrittenTypes.TypeDeclaration.Definition.Record recordFields)
                 |> Stdlib.Result.Result.Ok
 
+
+            | [ child ] when child.typ == "type_decl_def_enum" ->
+              let enumCases =
+                match findNodeByFieldName child "content" with
+                | Some contentNode ->
+                  contentNode.children
+                  |> Stdlib.List.map (fun caseNode ->
+                    let caseNameNode = findNodeByFieldName caseNode "case_name"
+
+                    let keywordOf =
+                      Stdlib.Option.map
+                        (findNodeByFieldName caseNode "keyword_of")
+                        (fun range -> range.sourceRange)
+
+                    let fields =
+                      caseNode.children
+                      |> Stdlib.List.filter (fun field ->
+                        field.typ == "type_decl_enum_field")
+                      |> Stdlib.List.map (fun field ->
+                        match findNodeByFieldName field "type" with
+                        | Some fieldTypeNode ->
+                          let label =
+                            (findNodeByFieldName field "identifier")
+                            |> Stdlib.Option.map (fun ln ->
+                              (ln.sourceRange, ln.text))
+
+                          let colonSymbol =
+                            (findNodeByFieldName field "symbol_colon")
+                            |> Stdlib.Option.map (fun ln -> ln.sourceRange)
+
+                          let fieldType = TypeReference.parse fieldTypeNode
+
+                          match fieldType with
+                          | Ok fieldType ->
+                            WrittenTypes.TypeDeclaration.EnumField
+                              { range = field.sourceRange
+                                typ = fieldType
+                                label = label
+                                description = ""
+                                symbolColon = colonSymbol }
+                          | Error _ ->
+                            (WrittenTypes.Unparseable { source = field })
+                        | None -> (WrittenTypes.Unparseable { source = field }))
+
+                    match caseNameNode with
+                    | Some caseNameNode ->
+                      (WrittenTypes.TypeDeclaration.EnumCase
+                        { range = caseNode.sourceRange
+                          name = (caseNameNode.sourceRange, caseNameNode.text)
+                          fields = fields
+                          description = ""
+                          keywordOf = keywordOf })
+                      |> Stdlib.Result.Result.Ok
+                    | None ->
+                      (WrittenTypes.Unparseable { source = caseNode })
+                      |> Stdlib.Result.Result.Error)
+                  |> Stdlib.Result.values
```

Comment:
I'm hesitant to use `Stdlib.Result.values` here because it ignores the errored values. Do you have any suggestions for a better approach?

Reply:
Yes, thanks!